pandas/core/panel.py

```diff
@@ -197,7 +199,7 @@ def _init_dict(self, data, axes, dtype=None):
             haxis = Index(ks)
 
         for k, v in compat.iteritems(data):
-            if isinstance(v, dict):
+            if isinstance(v, MutableMapping):
```

Comment:
It looks like we need this check because the line below can modify the argument passed into the constructor? That seems like an existing bug.